commit 75540216fa0e723f9dbcb6f7f2fbd5e3d9a3c00f
Author: Mark Salter <msalter@redhat.com>
Date:   Fri Mar 24 09:53:56 2017 -0400

    arm64: fix NULL dereference in have_cpu_die()
    
    commit 335d2c2d192266358c5dfa64953a4c162f46e464 upstream.
    
    Commit 5c492c3f5255 ("arm64: smp: Add function to determine if cpus are
    stuck in the kernel") added a helper function to determine if die() is
    supported in cpu_ops. This function assumes a cpu will have a valid
    cpu_ops entry, but that may not be the case for cpu0 is spin-table or
    parking protocol is used to boot secondary cpus. In that case, there
    is a NULL dereference if have_cpu_die() is called by cpu0. So add a
    check for a valid cpu_ops before dereferencing it.
    
    Fixes: 5c492c3f5255 ("arm64: smp: Add function to determine if cpus are stuck in the kernel")
    Signed-off-by: Mark Salter <msalter@redhat.com>
    Signed-off-by: Will Deacon <will.deacon@arm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/arm64/kernel/smp.c b/arch/arm64/kernel/smp.c
index 8507703..a70f7d3 100644
--- a/arch/arm64/kernel/smp.c
+++ b/arch/arm64/kernel/smp.c
@@ -934,7 +934,7 @@ static bool have_cpu_die(void)
 #ifdef CONFIG_HOTPLUG_CPU
 	int any_cpu = raw_smp_processor_id();
 
-	if (cpu_ops[any_cpu]->cpu_die)
+	if (cpu_ops[any_cpu] && cpu_ops[any_cpu]->cpu_die)
 		return true;
 #endif
 	return false;
